websites/A/AniWatch/presence.ts

```diff
@@ -0,0 +1,263 @@
+const presence = new Presence({
+		clientId: "1127962521609973881",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let data: {
+	currTime: number;
+	duration: number;
+	paused: boolean;
+} = null;
+
+presence.on(
+	"iFrameData",
+	async (recievedData: {
+		currTime: number;
+		duration: number;
+		paused: boolean;
+	}) => {
+		data = recievedData;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, href } = document.location,
+		buttons = await presence.getSetting<boolean>("buttons");
+
+	if (pathname === "/" || pathname === "/home")
+		presenceData.details = "Exploring AniWatch.to";
+	else if (
+		/\/(most-favorite|most-popular|movie|recently-added|recently-updated|tv|top-airing|top-upcoming|ona|ova|special|(genre\/.*))/.test(
+			pathname
+		)
+	) {
+		const heading = document.querySelector<HTMLHeadElement>("h2.cat-heading");
+		if (heading) presenceData.details = `Looking at ${heading.textContent}`;
+	} else if (pathname.startsWith("/news")) {
+		presenceData.details = "Looking at Anime news";
+		if (pathname !== "/news") {
+			const title = document.querySelector<HTMLHeadingElement>("h2.news-title");
+			if (title) presenceData.state = title.textContent;
+		}
+	} else if (pathname.startsWith("/community/user")) {
+		const profile = document.querySelector<HTMLSpanElement>(
+			"#main-wrapper > div.container > div > div.ai_-welcome.text-center > span"
+		);
+		if (profile) presenceData.details = `Viewing User: ${profile.textContent}`;
+		presenceData.smallImageKey = Assets.Viewing;
+	} else if (pathname.startsWith("/producer")) {
+		const name = document.querySelector(
+			"#main-content > section > div.block_area-header.block_area-header-tabs > div.float-left.bah-heading.mr-4 > h2"
+		)?.textContent;
+
+		presenceData.details = name;
+		if (buttons) {
+			presenceData.buttons = [
+				{
+					label: "Go to Page",
+					url: href,
+				},
+			];
+		}
+	} else if (
+		pathname.startsWith("/people") ||
+		pathname.startsWith("/character")
+	) {
+		const name = document.querySelector<HTMLHeadingElement>("h4.name");
+		if (name) {
+			presenceData.details = `Looking at ${
+				pathname.startsWith("/people") ? "People" : "Character"
```

Comment:
This may create grammatically incorrect phrases.
```suggestion
				pathname.startsWith("/people") ? "people" : "characters"
```
```suggestion
				pathname.startsWith("/people") ? "a person" : "a character"
```